src/price/service.price.platform.cpp

```diff
@@ -60,15 +61,38 @@ namespace antara::mmbot
         return st_price{price / nb_calls_succeed.load()};
     }
 
-    registry_price_result price_service_platform::get_price(const registry_quotes_for_specific_base &quotes_for_specific_base) const
+    registry_price_result price_service_platform::get_price(const registry_quotes_for_specific_base &quotes_for_specific_base)
     {
+        VLOG_SCOPE_F(loguru::Verbosity_INFO, pretty_function);
         antara::mmbot::registry_price_result res{};
         for (auto&&[current_symbol, current_quotes_table] : quotes_for_specific_base) {
             for (auto &&current_quote : current_quotes_table) {
                 antara::pair current_pair{antara::asset{current_quote}, antara::asset{st_symbol{current_symbol}}};
                 res.emplace(current_pair, this->get_price(current_pair));
             }
         }
+        all_price_.clear();
+        nlohmann::json all_price_json;
+        all_price_json["prices"] = nlohmann::json::array();
+        for (auto &&[current_pair, current_price] : res) {
+            auto current_object = nlohmann::json::object();
+            std::string current_price_str;
+            if (mmbot_config_.base_ercs_registry.at(current_pair.base.symbol.value())) {
+                current_price_str = fmt::format("{:.18f}", static_cast<double>(current_price.value()) / g_factor);
+            } else {
+                current_price_str = fmt::format("{:.8f}", static_cast<double>(current_price.value()) / g_factor);
```

Comment:
Can we put this logic somewhere we can reuse it? As a method of the asset struct?

Reply:
Yes good idea

Reply:
I realise that it's complicated by the `base_ercs_registry`, but maybe we can put this information inside the asset struct.

Reply:
the information about ercs is inside a configuration file: coins.json written by @jl777

That's why it's parsed once at the beginning of the program and can be used from everywhere